src/update-manager.ts

```diff
@@ -981,44 +997,19 @@ export default class UpdateManager {
 
     const documentString = this.serialize(doc.value, newSts, contentType)
 
-    // Write the new version back
-    // create component for file writing
-    // console.log('Writing back: <<<' + documentString + '>>>')
-    var filename = doc.value.slice(7) // chop off   file://  leaving /path
-    // console.log("Writeback: Filename: "+filename+"\n")
-    // @ts-ignore Where does Component come from? Perhaps deprecated?
-    var file = Components.classes[ '@mozilla.org/file/local;1' ]
-    // @ts-ignore Where does Component come from? Perhaps deprecated?
-      .createInstance(Components.interfaces.nsILocalFile)
-    file.initWithPath(filename)
-    if (!file.exists()) {
-      throw new Error('Rewriting file <' + doc.value +
-        '> but it does not exist!')
-    }
-    // {
-    // file.create( Components.interfaces.nsIFile.NORMAL_FILE_TYPE, 420)
-    // }
-    // create file output stream and use write/create/truncate mode
-    // 0x02 writing, 0x08 create file, 0x20 truncate length if exist
-    // @ts-ignore Where does Component come from? Perhaps deprecated?
-    var stream = Components.classes[ '@mozilla.org/network/file-output-stream;1' ]
-    // @ts-ignore Where does Component come from? Perhaps deprecated?
-      .createInstance(Components.interfaces.nsIFileOutputStream)
-
-    // Various JS systems object to 0666 in struct mode as dangerous
-    stream.init(file, 0x02 | 0x08 | 0x20, parseInt('0666', 8), 0)
-
-    // write data to file then close output stream
-    stream.write(documentString, documentString.length)
-    stream.close()
-
-    for (let i = 0; i < ds.length; i++) {
-      kb.remove(ds[ i ])
-    }
-    for (let i = 0; i < is.length; i++) {
-      kb.add(is[ i ].subject, is[ i ].predicate, is[ i ].object, doc)
-    }
-    callbackFunction(doc.value, true, '') // success!
+    kb.fetcher.webOperation('PUT',doc.value,{
+      "body"      : documentString,
+      contentType : contentType,
+    }).then( (response)=>{
+      if(!response.ok) return callbackFunction(doc.value,false,response.error)
+      for (var _i13 = 0; _i13 < ds.length; _i13++) {
```

Comment:
**suggestion**: Those variable names really confused me on first sight. I propose to change it back to `let i`